### ISSUE TITLE
Remove -D_WCHAR_T_DEFINED from scripts and parser profiles

### ISSUE THREAD

**Issue by xiaoyinl** [NationalSecurityAgency/ghidra#5196]
**Describe the bug**
`wchar_t` has been removed as a keyword in 8c71f9d7fabd4ca2c540cd24e5e6272682810f6c (#5108). `-D_WCHAR_T_DEFINED` and `-D_WCHAR_T` should be removed from the following files:
```
Ghidra/Processors/JVM/ghidra_scripts/CreateJNIGDTArchivesScript.java:116:                               "-D_WCHAR_T",
Ghidra/Processors/JVM/ghidra_scripts/CreateJNIGDTArchivesScript.java:162:                               "-D_WCHAR_T",
Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java:476:                             "-D_WCHAR_T_DEFINED",
Ghidra/Features/Base/data/parserprofiles/MacOSX_10.5.prf:210:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/generic_clib_64.prf:234:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/linux_32.prf:531:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/MacOSX_10.9.prf:326:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf:196:-D_WCHAR_T_DEFINED
Ghidra/Features/Base/data/parserprofiles/VisualStudio9.prf:88:-D_WCHAR_T_DEFINED
Ghidra/Features/Base/data/parserprofiles/vs12Local.prf:242:-D_WCHAR_T_DEFINED
Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf:194:-D_WCHAR_T_DEFINED
Ghidra/Features/Base/data/parserprofiles/generic_clib_32.prf:234:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/linux_64.prf:531:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/MacOSX_Cocoa.prf:31:-D_WCHAR_T
Ghidra/Features/Base/data/parserprofiles/clib.prf:90:-D_WCHAR_T
```
This list is generated by `grep -n -r 'D_WCHAR_T'`.

**Environment (please complete the following information):**
 - Ghidra Version: 10.3 75a185aa9efc65f98b2291c9affaeb92d6163cfd

**Comment by emteere** [comment]
Agree they should be removed.  The changes were tested to make sure the C parsing can handle a header file defining wchar_t with a typedef or relying on the it already defined internally by the compiler.

wchar_t is replaced with the Ghidra built-in wchar_t to better handle cross data organization conversions.

**Comment by xiaoyinl** [comment]
> My hope is you aren't seeing any errors in parsing because of the left in -D_WCHAR_T or -D_WCHAR_T_DEFINED entries noted above?

No, but I haven't used the C Parser after the recent change. I opened this issue because I happened to read the commit 8c71f9d7fabd4ca2c540cd24e5e6272682810f6c, and found that `-D_WCHAR_T` and `-D_WCHAR_T_DEFINED` were removed from CreateDefaultGDTArchivesScript.java but not other files.

**Comment by xiaoyinl** [comment]
fixed in 2cdaebf0c350ad10bb005d26a9cdb73cc9d1ba94
